utils/keys_test.go

```diff
@@ -298,20 +401,153 @@ func TestImportKeys2InOneFile(t *testing.T) {
 
 	in := bytes.NewBuffer(byt)
 
-	err := ImportKeys(in, []Importer{s})
+	err := ImportKeys(in, []Importer{s}, "", "", passphraseRetriever)
 	require.NoError(t, err)
 
 	bFinal, bRest := pem.Decode(s.data["ankh"])
 	require.Equal(t, b.Bytes, bFinal.Bytes)
-	require.Len(t, bFinal.Headers, 0) // path header is stripped during import
+	_, ok := bFinal.Headers["path"]
+	require.False(t, ok, "expected no path header, should have been removed at import")
+	role, _ := bFinal.Headers["role"]
+	require.Equal(t, notary.DefaultImportRole, role)
 
 	b2Final, b2Rest := pem.Decode(bRest)
 	require.Equal(t, b2.Bytes, b2Final.Bytes)
-	require.Len(t, b2Final.Headers, 0) // path header is stripped during import
+	_, ok = b2Final.Headers["path"]
+	require.False(t, ok, "expected no path header, should have been removed at import")
 	require.Len(t, b2Rest, 0)
 
 	cFinal, cRest := pem.Decode(s.data["morpork"])
 	require.Equal(t, c.Bytes, cFinal.Bytes)
-	require.Len(t, cFinal.Headers, 0)
+	_, ok = cFinal.Headers["path"]
+	require.False(t, ok, "expected no path header, should have been removed at import")
 	require.Len(t, cRest, 0)
 }
+
+func TestImportKeys2InOneFileNoPath(t *testing.T) {
+	s := NewTestImportStore()
+
+	from, _ := os.OpenFile("../fixtures/secure.example.com.key", os.O_RDONLY, notary.PrivKeyPerms)
+	defer from.Close()
+	fromBytes, _ := ioutil.ReadAll(from)
+	b, _ := pem.Decode(fromBytes)
+	b.Headers["role"] = data.CanonicalSnapshotRole
+	bBytes := pem.EncodeToMemory(b)
+
+	b2, _ := pem.Decode(fromBytes)
+	b2.Headers["role"] = data.CanonicalSnapshotRole
+	b2Bytes := pem.EncodeToMemory(b2)
+
+	c := &pem.Block{
+		Headers: make(map[string]string),
+	}
+	c.Bytes = make([]byte, 1000)
+	rand.Read(c.Bytes)
+	c.Headers["path"] = "morpork"
+
+	bBytes = append(bBytes, b2Bytes...)
+	cBytes := pem.EncodeToMemory(c)
+
+	byt := append(bBytes, cBytes...)
+
+	in := bytes.NewBuffer(byt)
+
+	err := ImportKeys(in, []Importer{s}, "", "", passphraseRetriever)
+	require.NoError(t, err)
+
+	bFinal, bRest := pem.Decode(s.data[filepath.Join(notary.NonRootKeysSubdir, "12ba0e0a8e05e177bc2c3489bdb6d28836879469f078e68a4812fc8a2d521497")])
```

Comment:
I'm okay with either method, but I would be tempted to lean towards not importing it since we don't import encrypted keys without a path etc (cases where the user messed up as well). Would make sense to have some consistency there. This would mean changing the code and this test etc since as @cyli mentioned correctly we now just import the key. Happy to do it once we decide @riyazdf @endophage